### PR TITLE
add -x -p -h args so ldmsd_controller options are like others

### PR DESCRIPTION
**Pull request by baallan**
changes -h (or --help) to -? or --help to remedy collision with our -h host common syntax.

### CODE REVIEW

File: ldms/python/ldmsd/ldmsd_controller [hunk 1]
```diff
@@ -1712,13 +1712,16 @@ if __name__ == "__main__":
                                          "To connect to an ldmsd, either give " \
                                          "the socket path of the ldmsd or " \
                                          "give both hostname and inet control port. " \
-                                         "If all are given, the sockname takes the priority.")
-        parser.add_argument('--host',
+                                         "If all are given, the sockname takes the priority.",
+                                         add_help=False)
+        parser.add_argument('-H', '--help', action='help', default='==SUPPRESS==',
```
Comment (tom95858) [inline ldms/python/ldmsd/ldmsd_controller:1717]: I think '?' might be better than -H. @nichamon, what do you think?

Comment (baallan) [reply ldms/python/ldmsd/ldmsd_controller:1717]: I'll update it to any concensus value. I think -? is somewhat less common than -H in the python world, but i can't point to a survey to prove it. Could probably add both, in fact, unless you have a collision with -H in mind for a feature down the road.

Comment (nichamon) [reply ldms/python/ldmsd/ldmsd_controller:1717]: > I think '?' might be better than -H. @nichamon, what do you think?

If we want to have a shorthand for --help, I vote for -?. I think we won't use -? for any other arguments. Many programs use -H for a particular argument. For example, docker's -H means a path to a file containing a host list. salloc's -H is equivalent to --hold.

Comment (tom95858) [reply ldms/python/ldmsd/ldmsd_controller:1717]: @baallan, Ok, I like this patch, but can we go with -??

Comment (baallan) [reply ldms/python/ldmsd/ldmsd_controller:1717]: Will update with the ? instead of H.

File: ldms/man/ldmsd_controller.man [hunk 2]
```diff
@@ -64,7 +64,7 @@ Path to the config file
 .br
 Execute the script and send the output commands to the connected ldmsd
 .TP
-.BR -h
+.BR -H
```
Comment (tom95858) [inline ldms/man/ldmsd_controller.man:67]: This needs to be -? right?

Comment (baallan) [reply ldms/man/ldmsd_controller.man]: yes, fixed.

**Comment by tom95858** [comment]
@baallan could you please squash these 4 commits into one and then force push to your branch. If I squash it here it creates a garbage commit message.

**Comment by baallan** [comment]
@tom95858 squashed as requested.

**Pull request merged**